### ISSUE TITLE
Pinned tab highlighting not working

### ISSUE THREAD

**Issue by VittGam** [Aris-t2/ClassicThemeRestorer#185]
Hello,

I'm having an issue with tab highlighting when the title of a pinned tab changes, when using the classic square tab theme (the second option in the dropdown menu).

![ffbug](https://cloud.githubusercontent.com/assets/615188/14440628/1317c15a-0031-11e6-841a-48f036c414ae.png)

The `[titlechanged]:not(:hover)` state is bad here.

- Classic Theme Restorer v1.5.0
- Firefox/Iceweasel 44 with default theme
- Arch Linux, Plasma 5 and Oxygen GTK theme

Cheers,
Vittorio

**Comment by Aris-t2** [comment]
Thanks for reporting.

Could you test CTR beta 1.5.1 b1?
https://addons.mozilla.org/addon/classicthemerestorer/versions/1.5.1beta1

**Comment by VittGam** [comment]
You're welcome!

The titlechanged tab button now has the same background as in the default Firefox theme.

When hovering, it has both the default background, and the style 2 blue color.

I think that style 1 should have the same backgrounds as style 2, it seems more consistent overall.

![ffbug2](https://cloud.githubusercontent.com/assets/615188/14469865/e5c2d62a-00e6-11e6-9a71-4d5d17e51c79.png)

**Comment by Aris-t2** [comment]
You are right, it already is like this on Windows, but I missed to add this to Linux too.
Next version will resolve this problem.

**Comment by Aris-t2** [comment]
Is tab highlighting on your Linux configuration fine again with latest CTR beta?

**Comment by VittGam** [comment]
I don't think it's fine...

![ffbug3](https://cloud.githubusercontent.com/assets/615188/14568505/913606e0-0339-11e6-8ad9-413f00e1ec12.png)

**Comment by Aris-t2** [comment]
Why not?

Fx 28 - normal = CTRs squared tabs v1
![1](https://cloud.githubusercontent.com/assets/6013569/14570893/d8ba794e-0346-11e6-8220-17b9ccd6f654.png)
Fx 28 - hover = CTRs squared tabs v1

CTR uses same colors cross platform for "squared tabs v1".


Fx 48 - normal - CTRs squared tabs v2
![2](https://cloud.githubusercontent.com/assets/6013569/14571205/342584a8-0348-11e6-83b9-8161bd2ccb70.png)
Fx 48 - hover - CTRs squared tabs v2

CTRs squared tabs v2 are not meant to match v1 colors. They are slightly different on purpose.

**Comment by VittGam** [comment]
There is the "modern" background (a dot with a blurred border positioned in the bottom center of the button) applied along with the good old background (the light shadow positioned at the top center of the button).

It was the beta1 that introduced the wrong background.

**Comment by Aris-t2** [comment]
You reported a bug with "titlechanged"-status of pinned tabs. Checked Fx 28 on Linux and Windows made me realize pinned tabs code needs adjustments, which got passed to CTR 1.5.1 beta.

Compare yourself (screenshots taken on Linux Mint 17) --->

Fx 28, titlechanged tab, normal - no CTR
![fx28](https://cloud.githubusercontent.com/assets/6013569/14579963/1366e02e-03be-11e6-8cdb-d255fb891575.png)
Fx 28, titlechanged tab, hover - no CTR

-----------------------------------

Fx 45, titlechanged tab, normal - CTR 1.5.0 (buggy)
![fx45_ctr150](https://cloud.githubusercontent.com/assets/6013569/14579964/197ce1e8-03be-11e6-8438-2f1be02cff24.png)
Fx 45, titlechanged tab, hover - CTR 1.5.0 (buggy)

-----------------------------------

Fx 45, titlechanged tab, normal - CTR 1.5.1 beta 4 (fixed)
![fx45_ctr151b](https://cloud.githubusercontent.com/assets/6013569/14579965/1cfe223c-03be-11e6-8abe-dbe3a700e953.png)
Fx 45, titlechanged tab, hover - CTR 1.5.1 beta 4 (fixed)

-----------------------------------

I think you will agree pinned titlechanged tab colors with latest CTR beta are much closer to original Fx 28 colors than the (bugged) ones CTR 1.5.0 provides atm. The "modern" code you are referring to is used for "squared tabs (australized) v2" and "curved tabs (alternative)" while "squared tabs (australized)" and "squared tabs (classic) v2" use something between "squared tabs (classic)" and "curved tabs (alternative)".

To sum up: CTR 1.5.1 beta 4 + ...
...squared tabs (classic) : classic background
...squared tabs (classic) v2: mix between classic and modern background, more classic
...squared tabs (australized): mix between classic and modern background, more modern
...squared tabs (australized) v2: modern background
...curved tabs (alternative): modern background

**Comment by VittGam** [comment]
Yes, now it's better than before. But the bad "modern" indicator can be actually removed by uncommenting this line again:

https://github.com/Aris-t2/ClassicThemeRestorer/commit/cc48c20f357552804124d4350f50cd9057952ca0#diff-d2c85c17970a7675365aaa0b86f02b02R47

I just tried with Stylish and it works.

**Comment by Aris-t2** [comment]
CTR 1.5.1 beta 5 will simplify "titlechanged" status cross platform like this:

squared tabs (classic): classic background
squared tabs (classic) v2: classic background
squared tabs (australized): modern indicator
squared tabs (australized) v2: modern indicator
curved tabs (alternative): modern indicator

**Comment by VittGam** [comment]
It's now fixed. Thank you very much!
